storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,711 @@
+        } else if (key.endsWith("-stormcode.ser")) {
+            ret = key.substring(0, key.length() - 14);
+        } else if (key.endsWith("-stormconf.ser")) {
+            ret = key.substring(0, key.length() - 14);
+        }
+        return ret;
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedWorkerUserWSE mocked = new SetMockedWorkerUserWSE(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedWorkerUserWSE. conf)]
+    //     run test ...)
+    public static class SetMockedWorkerUserWSE implements Closeable {
+        public SetMockedWorkerUserWSE(Map conf) {
+            mockedWorkerUserWSE = conf;
+        }
+
+        @Override
+        public void close() {
+            mockedWorkerUserWSE = null;
+        }
+    }
+    private static Map mockedWorkerUserWSE = null;
+    public static void setWorkerUserWSE(Map conf, String workerId, String user) throws IOException {
+        if (mockedWorkerUserWSE != null) {
+            return;
+        }
+        LOG.info("SET worker-user {} {}", workerId, user);
+        File file = new File(workerUserFile(conf, workerId));
+        file.getParentFile().mkdirs();
+
+        try (FileWriter fw = new FileWriter(file);
+             BufferedWriter writer = new BufferedWriter(fw);) {
+            writer.write(user);
+        }
+    }
+
+    public static void removeWorkerUserWSE(Map conf, String workerId) {
+        LOG.info("REMOVE worker-user {}", workerId);
+        new File(workerUserFile(conf, workerId)).delete();
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedWorkerArtifactsRoot mocked = new SetMockedWorkerArtifactsRoot(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedWorkerArtifactsRoot. root)]
+    //     run test ...)
+    public static class SetMockedWorkerArtifactsRoot implements Closeable {
+        public SetMockedWorkerArtifactsRoot(String root) {
+            mockedWorkerArtifactsRoot = root;
+        }
+
+        @Override
+        public void close() {
+            mockedWorkerArtifactsRoot = null;
+        }
+    }
+    private static String mockedWorkerArtifactsRoot = null;
+    public static String workerArtifactsRoot(Map conf) {
+        if (mockedWorkerArtifactsRoot != null) {
+            return mockedWorkerArtifactsRoot;
+        }
+        String artifactsDir = (String)conf.get(Config.STORM_WORKERS_ARTIFACTS_DIR);
+        if (artifactsDir == null) {
+            return (LOG_DIR + FILE_SEPARATOR + "workers-artifacts");
+        } else {
+            if (new File(artifactsDir).isAbsolute()) {
+                return artifactsDir;
+            } else {
+                return (LOG_DIR + FILE_SEPARATOR + artifactsDir);
+            }
+        }
+    }
+
+    public static String workerArtifactsRoot(Map conf, String id) {
+        if (mockedWorkerArtifactsRoot != null) {
+            return mockedWorkerArtifactsRoot;
```

Comment:
I'm not sure how the mocking worked in the clj code.  This looks OK, but is confusing because we are not including the ID.  It might be worth a comment here on why that is.